stanage julia docs (#1965)
rcgsheffield/sheffield_hpc: stanage/software/apps/julia.rst

.. _julia_stanage:

.. |softwarename| replace:: Julia
.. |currentver| replace:: 1.9.0

|softwarename|
==============

.. sidebar:: |softwarename|

   :Versions:   |currentver|
   :URL: https://julialang.org/
   :Documentation:  https://docs.julialang.org/en/v1/

>    :Dependencies: EasyBuild (see :ref:`stanage_eb_toolchains`).

Reviewer: Isn't it a binary install with no run-time dependencies?  Not dependent on any EB toolchains anyways.

The Julia programming language is a flexible dynamic language, appropriate for scientific and numerical computing, with performance comparable to traditional statically-typed languages. For more information visit: https://docs.julialang.org/en/v1/  

Interactive Usage
-----------------
After connecting to Stanage,  start an interactive session with the ``srun --pty bash –i`` command.

> The latest version of MATLAB (currently 2022a) is made available by running:

Reviewer: Julia
```suggestion
Load a particular version of Julia with:
```

.. code-block:: bash

>    module Julia/1.9.0-linux-x86_64

Reviewer 2: ```suggestion
   module load Julia/1.9.0-linux-x86_64
```

You can then start Julia with ``julia``.

Batch Usage
-----------
Here, we assume that you wish to run the program ``example.jl`` on the system:

.. code-block:: bash

    result = 0
  
    # Prompt to enter 
    println("Enter 5 numbers line by line") 
    
    # Taking Input from user 
    for number in 1:5 
    
    num = readline() 
    num = parse(Int64, num)  
    global result+= num   
    
    end 
    
    println("The sum is :", result) 

First, you need to write a batch submission file. We assume you’ll call this ``my_job.slurm``:   

.. code-block:: bash

    #!/bin/bash
    #SBATCH --ntasks=1
    #SBATCH --time=10:00

>     #SBATCH --mem-per-cpu=100

Reviewer: ```suggestion
    #SBATCH --mem=100
```
Simpler?

    
    #load the julia module
    module load Julia/1.9.0-linux-x86_64

    julia example.jl

> Ensure that ``example.jl`` and ``my_job.slurm`` are both in your current working directory, then submit your job to the batch system:

Reviewer 2: ```suggestion
Ensure that ``example.jl`` and ``my_job.slurm`` are both in your current working directory, then submit your job to the SLURM scheduler:
```

.. code-block:: bash
    
    sbatch my_job.slurm

Installation notes
------------------

> |softwarename| version 1.9.0 was installed using Easybuild 4.7.0, build details can be found 

Reviewer: NCDU?

> |softwarename| version 1.9.0 was installed using Easybuild 4.7.0, build details can be found 

Reviewer: ```suggestion
|softwarename| version 1.9.0 was installed using Easybuild 4.8.1, build details can be found 
```

> in ``$EBROOTNCDU/easybuild`` with the module loaded.

Reviewer: ```suggestion
in ``$EBROOTJULIA/easybuild`` with the module loaded.
```
